pulser-core/pulser/json/abstract_repr/deserializer.py

```diff
@@ -236,16 +239,46 @@ def deserialize_abstract_sequence(obj_str: str) -> Sequence:
     device_name = obj["device"]
     device = getattr(devices, device_name)
 
+    # Register Layout
+    layout = (
+        RegisterLayout(obj["layout"]["coordinates"])
+        if "layout" in obj
+        else None
+    )
+
     # Register
+    reg: Union[BaseRegister, MappableRegister]
     qubits = obj["register"]
-    reg = Register({q["name"]: (q["x"], q["y"]) for q in qubits})
+    if {"name", "x", "y"} == qubits[0].keys():
```

Comment:
I think it would have to be a field that is only available (and required) when the register is mappable (I'm thinking something like `with_mappable_reg: true`). 
Otherwise, it would be a breaking change unless it is optional (which would defeat the purpose). 
Should we go for it?

Reply:
Arf, true. 
In this case, it's pretty easy so let's keep it this way. But next time we need to do `oneOf`s, using a `type` prop` can be very helpful.